src/main/java/com/yahoo/memory/BaseWritableMemoryImpl.java

```diff
@@ -168,21 +169,21 @@ private static void copyMemorySlowPath(final Object srcUnsafeObj, final long src
 
   static void copyMemoryCheckingNonOverlapping(final Object srcUnsafeObj, final long srcAdd,
       final Object dstUnsafeObj, final long dstAdd, final long lengthBytes) {
-    if (srcUnsafeObj != dstUnsafeObj || srcAdd + lengthBytes <= dstAdd ||
-        dstAdd + lengthBytes <= srcAdd) {
+    if ((srcUnsafeObj != dstUnsafeObj) || ((srcAdd + lengthBytes) <= dstAdd)
+        || ((dstAdd + lengthBytes) <= srcAdd)) {
       copyNonOverlappingMemory(srcUnsafeObj, srcAdd, dstUnsafeObj, dstAdd, lengthBytes);
     } else {
-      throw new IllegalArgumentException("Not expecting memory blocks to overlap: obj=" +
-          srcUnsafeObj + ", srcAddress=" + srcAdd + ", dstAddress=" + dstAdd + ", lengthBytes=" +
-          lengthBytes);
+      throw new IllegalArgumentException("Not expecting memory blocks to overlap: obj="
+          + srcUnsafeObj + ", srcAddress=" + srcAdd + ", dstAddress=" + dstAdd
+          + ", lengthBytes=" + lengthBytes);
     }
   }
 
-  /** @see #UNSAFE_COPY_MEMORY_THRESHOLD */
+  /* @see #UNSAFE_COPY_MEMORY_THRESHOLD */
```

Comment:
Why turning a javadoc comment into a non-javadoc comment?

Reply:
1) As a javadoc comment it was very incomplete.  It was missing documentation of all 5 parameters. Because of this it raised 5 warnings in my IDE.  You need to have similar detections set up in your IDE.
2) The @see is a reference, which should be used as additional information after an english sentence that explains the intent of the method. The @see did not explain the intent of the method, it is more of a code comment than a javadoc comment.
3) It is a private method.  Javadocs are always good, but not required for private methods.  If you want to write a javadoc, then make sure it is at least a complete javadoc.

Reply:
I made this a Javadoc method to enable easy navigation in the IDE (Ctrl + click). The method and the constant have the common description, so I put it on the constant's Javadoc, and the method's one just links to it.

Yes, because this method is private, all those strict Javadoc lint rules shouldn't necessarily be followed. It's purpose just to easy reading, understanding, navigation for the code developers.

Reply:
But it's not that important, it was just a question. Code could stay the same

Reply:
I don't know how to set more relaxed lint rules for private methods.  In my eclipse and checkstyle, a javadoc is not required for private methods.  But once you start a javadoc with /** all the lint rules apply.  

When I first open a piece of code, one of the first things I do is check all the warnings.  If they are easy to fix, I usually just fix them and go on.  This one, however, was harder to fix properly, and I didn't think that changing it to a code comment lost any information, since private javadocs don't appear in the public javadocs, the only  way someone would see this would be when they are looking at the code.